### ISSUE TITLE
v0.10.0 causes runtime error when building with Create React App

### ISSUE THREAD

**Issue by pangolingo** [vitalets/websocket-as-promised#21]
I'm using websocket-as-promised in an ejected version of "create-react-app". When testing using the dev server, everything works fine. But when I build the project and then run it, I get an error "Uncaught TypeError: Super expression must either be null or a function, not undefined" and the websocket code fails to run.

Downgrading to version 0.9.0 fixes the problem.

I have not had a chance to try much debugging. I tried to change the `Terser` minification plugin option in my app to `keep_fnames: true`, going on advice from various Stack Overflow posts. This did not resolve the issue. I also tried disabled the `Terser` plugin in my app, and that resolved the issue.

**Comment by vitalets** [comment]
hi @pangolingo !
Could you show a stack trace to localize the problem?

**Comment by pangolingo** [comment]
Unfortunately the stack trace is almost useless because I didn't have time to fight with Webpack to include the package sourcemaps. The root error comes from line 1065 in `dist/index.js`: `  if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a `

```
(anonymous) (index.js:1065)
function");
(anonymous) (index.js:1064)
(anonymous) (index.js:1058)
(anonymous) (index.js:711)
n (index.js:15)
(anonymous) (index.js:454)
n (index.js:15)
(anonymous) (index.js:51)
226 (index.js:5)
f (index.html:1)
486 (index.jsx:15)
f (index.html:1)
229 (index.js:1924)
f (index.html:1)
t (index.html:1)
r (index.html:1)
(anonymous) (index.chunk.js:1)
```

The error happens when I instantiate the library:
```
import WebSocketAsPromised from 'websocket-as-promised';

const client = new WebSocketAsPromised(`ws://localhost:${port}`, {
      packMessage: (data: any) => JSON.stringify(data),
      unpackMessage: (data: any) => data,
      // add a unique ID to the request so we know when we receive a response
      attachRequestId: (data: any, requestId: string | number): void => {
        console.log('attaching id', requestId, 'to', data);
        if (
          typeof data === 'object' &&
          'rc' in data &&
          typeof data.rc === 'object'
        ) {
          const newData = { ...data };
          newData.rc.id = requestId;
          console.log('attached', newData);
          return newData;
        }
        console.log('could not attach');
        return data;
      },
      extractRequestId: (data: any) => {
        if (
          typeof data === 'object' &&
          'rc' in data &&
          typeof data.rc === 'object'
        ) {
          return data.rc.id;
        }
        return null;
      }
    });
```

**Comment by pangolingo** [comment]
I just confirmed this failure occurs with a fresh `create-react-app` installation using CRA v3.0.1.

Steps to reproduce:
```
> create-react-app 
> cd websocket-as-promised-error-poc
```
Add the following to the top of `App.js`:
```
import WebSocketAsPromised from 'websocket-as-promised';
new WebSocketAsPromised('ws://localhost:8080');
```
Build and run the app (I'm using `http-server` to run it.)
```
> yarn build
> http-server ./build
```

**Comment by vitalets** [comment]
Thanks for details!
I've disabled own minification and released 0.10.1.
In my tests with CRA it now works.

Could you please reinstall fresh websocket-as-promised and re-check?

**Comment by pangolingo** [comment]
@vitalets I've confirmed that the bug is fixed in 0.10.1. Thank you! 🙇‍♂️

**Comment by vitalets** [comment]
You are welcome! )
